src/lib/mdns/Resolver_ImplMinimalMdns.cpp

```diff
@@ -181,7 +184,9 @@ void PacketDataReporter::OnDiscoveredNodeIPAddress(const chip::Inet::IPAddress &
     {
         return;
     }
-    mDiscoveredNodeData.ipAddress[mDiscoveredNodeData.numIPs++] = addr;
+    mDiscoveredNodeData.ipAddress[mDiscoveredNodeData.numIPs]   = addr;
+    mDiscoveredNodeData.interfaceId[mDiscoveredNodeData.numIPs] = mInterfaceId;
+    mDiscoveredNodeData.numIPs++;
```

Comment:
Same here: how do we guarantee bound checks on the numIPs argument

Reply:
There is a check before these assignments at line 182.